infra/build_specified_commit.py

```diff
@@ -50,7 +51,13 @@ def build_fuzzer_from_commit(project_name,
     0 on successful build 1 on failure
   """
   if not old_repo_manager:
```

Comment:
I missed this in a previous review, but this logic is a bit confusing. Let's just require callers to always provide a `repo_manager`. That way we don't need to add this `repo_name` argument either and complicate this function more.

Doing a grep it seems like it's only tests that don't pass this argument anyway.

```
git grep -A5  build_fuzzer_from_commit
bisector.py:    build_specified_commit.build_fuzzer_from_commit(
bisector.py-        build_data.project_name, commit_list[new_idx],
bisector.py-        bisect_repo_manager.repo_dir, build_data.engine, build_data.sanitizer,
bisector.py-        build_data.architecture, bisect_repo_manager)
bisector.py-    expected_error_code = helper.reproduce_impl(build_data.project_name,
bisector.py-                                                fuzz_target, False, [], [],
--
bisector.py:    build_specified_commit.build_fuzzer_from_commit(
bisector.py-        build_data.project_name, commit_list[old_idx],
bisector.py-        bisect_repo_manager.repo_dir, build_data.engine, build_data.sanitizer,
bisector.py-        build_data.architecture, bisect_repo_manager)
bisector.py-    oldest_error_code = helper.reproduce_impl(build_data.project_name,
bisector.py-                                              fuzz_target, False, [], [],
--
bisector.py:      build_specified_commit.build_fuzzer_from_commit(
bisector.py-          build_data.project_name, commit_list[curr_idx],
bisector.py-          bisect_repo_manager.repo_dir, build_data.engine, build_data.sanitizer,
bisector.py-          build_data.architecture, bisect_repo_manager)
bisector.py-      error_code = helper.reproduce_impl(build_data.project_name, fuzz_target,
bisector.py-                                         False, [], [], testcase)
--
build_specified_commit.py:def build_fuzzer_from_commit(project_name,
build_specified_commit.py-                             commit,
build_specified_commit.py-                             local_store_path,
build_specified_commit.py-                             engine='libfuzzer',
build_specified_commit.py-                             sanitizer='address',
build_specified_commit.py-                             architecture='x86_64',
--
build_specified_commit_test.py:      build_specified_commit.build_fuzzer_from_commit(
build_specified_commit_test.py-          project_name, old_commit, tmp_dir, sanitizer='address')
build_specified_commit_test.py-      old_error_code = helper.reproduce_impl(project_name, fuzzer, False, [],
build_specified_commit_test.py-                                             [], test_data)
build_specified_commit_test.py:      build_specified_commit.build_fuzzer_from_commit(
build_specified_commit_test.py-          project_name, new_commit, tmp_dir, sanitizer='address')
build_specified_commit_test.py-      new_error_code = helper.reproduce_impl(project_name, fuzzer, False, [],
build_specified_commit_test.py-                                             [], test_data)
build_specified_commit_test.py-      self.assertNotEqual(new_error_code, old_error_code)
build_specified_commit_test.py
```

Reply:
Done.